### PR TITLE
Add base argument class for conditionally reading objects.

### PR DESCRIPTION
**Pull request by balamurugana**
`GetObject`, `HeadObject` and `CopyObject` S3 API support conditions
on reading objects. This patch introduces common base class
`ObjectConditionalReadArgs` for `GetObjectArgs`, `StatObjectArgs`,
`CopySource` and `ComposeSource` to avoid code duplication.

### CODE REVIEW

File: api/src/main/java/io/minio/StatObjectArgs.java [hunk 1]
```diff
@@ -17,11 +17,28 @@
 package io.minio;
 
 /** Argument class of MinioClient.statObject(). */
-public class StatObjectArgs extends ObjectReadArgs {
+public class StatObjectArgs extends ObjectConditionalReadArgs {
+  protected StatObjectArgs() {}
+
+  public StatObjectArgs(ObjectConditionalReadArgs args) {
```
Comment (sinhaashish) [inline api/src/main/java/io/minio/StatObjectArgs.java:23]: As the name suggest  `StatObjectArgs` is an builder type then why is the constructor used. IMO it should be renamed to `StatObject` similar to `ComposeObject`.

File: api/src/main/java/io/minio/ComposeSource.java [hunk 2]
```diff
@@ -115,90 +109,36 @@ public void buildHeaders(long objectSize, String etag) throws IllegalArgumentExc
...
-    }
-
-    public Builder length(long length) {
-      validateNullOrPositive(length, "length");
-      operations.add(args -> args.length = length);
-      return this;
+  public long objectSize() throws InternalException {
+    if (this.objectSize == null) {
+      throw new InternalException(
+          "buildHeaders(long objectSize, String etag) must be called prior this method invocation");
```
Comment (kannappanr) [inline api/src/main/java/io/minio/ComposeSource.java:122]:
```suggestion
          "buildHeaders(long objectSize, String etag) must be called prior to the invocation of this method");
```

File: api/src/main/java/io/minio/ObjectConditionalReadArgs.java [hunk 3]
```diff
@@ -0,0 +1,143 @@
+/*
+ * MinIO Java SDK for Amazon S3 Compatible Cloud Storage,
+ * (C) 2019 MinIO, Inc.
```
Comment (kannappanr) [inline api/src/main/java/io/minio/ObjectConditionalReadArgs.java:3]:
```suggestion
 * (C) 2020 MinIO, Inc.
```

File: api/src/main/java/io/minio/ComposeSource.java (hunk 2 above)
Comment (balamurugana) [reply api/src/main/java/io/minio/ComposeSource.java:122]: added `to`

File: api/src/main/java/io/minio/ObjectConditionalReadArgs.java (hunk 3 above)
Comment (balamurugana) [reply api/src/main/java/io/minio/ObjectConditionalReadArgs.java:3]: This file is there since 2019, now added 2020

Comment (balamurugana) [reply api/src/main/java/io/minio/ObjectConditionalReadArgs.java:3]: done

**Pull request merged**